chef/cookbooks/crowbar/templates/default/crowbar-pre-upgrade.sh.erb

```diff
@@ -54,4 +54,26 @@ ovs-vsctl set-fail-mode <%= bridge %> standalone
 echo "Reloading wicked configuration for all interfaces"
 wicked ifreload all
 
+<% if @compute_node %>
+
+echo "Upgrading services needed to act as nova-compute node"
+
+zypper --non-interactive up openstack-nova-compute
+zypper --non-interactive up openstack-neutron-openvswitch-agent
+zypper --non-interactive up python-pecan
+zypper --non-interactive install crudini
+zypper --non-interactive up openstack-cinder-volume
+
+echo "Upgrading configuration for Newton based packages"
+
+crudini --set /etc/neutron/plugins/ml2/openvswitch_agent.ini ovs ovsdb_interface vsctl
+crudini --set /etc/neutron/plugins/ml2/openvswitch_agent.ini ovs of_interface ovs-ofctl
+crudini --set /etc/nova/nova.conf DEFAULT use_neutron true
+
+systemctl start openstack-nova-compute.service
+systemctl start openstack-neutron-openvswitch-agent.service
```

Comment:
Hm, which agent(s) to start here really depends on the neutron proposal. E.g. it might also be linuxbridge. And in certain setups (DVR) even l3-agent (and maybe others are needed).